ansible_ai_connect/ai/api/model_pipelines/wca/pipelines_saas.py

```diff
@@ -363,6 +363,9 @@ def invoke(self, params: RoleGenerationParameters) -> RoleGenerationResponse:
         if outline:
             data["outline"] = outline
 
+        if not settings.ANSIBLE_AI_ENABLE_ROLE_GEN_ENDPOINT:
```

Comment:
IDK if it'd be clearer to move this closer to the start of the function?

e.g. https://github.com/ansible/ansible-ai-connect-service/blob/67eadca907c4068e2b35cdc6c26c20ccbf13a508/ansible_ai_connect/ai/api/model_pipelines/wca/pipelines_saas.py#L325-L329

The function performs some (potentially _expensive_) operations that are thrown away if not enabled.